app/src/test/java/com/codesoom/assignment/application/ProductServiceTest.java

```diff
@@ -0,0 +1,248 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.common.ProductFactory;
+import com.codesoom.assignment.common.exception.ProductNotFoundException;
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.domain.ProductCommand;
+import com.codesoom.assignment.domain.ProductInfo;
+import com.codesoom.assignment.domain.ProductRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+@DisplayName("ProductService 클래스")
+class ProductServiceTest {
+
+    private ProductService productService;
+
+    private ProductRepository productRepository;
+
+    @BeforeEach
+    void setUp() {
+        productRepository = mock(ProductRepository.class);
+        productService = new ProductServiceImpl(productRepository);
+    }
+
+    @Nested
+    @DisplayName("getProducts 메소드는")
+    class Describe_getProducts {
+        @Nested
+        @DisplayName("데이터가 존재한다면")
+        class Context_with_non_empty_data {
+            private final List<Product> givenProducts = new ArrayList<>();
+
+            @BeforeEach
+            void prepare() {
+
+                final Product product1 = ProductFactory.createProduct(1L);
+                final Product product2 = ProductFactory.createProduct(2L);
+
+                givenProducts.add(product1);
+                givenProducts.add(product2);
```

Comment:
```suggestion
                givenProducts.add(ProductFactory.createProduct(1L));
                givenProducts.add(ProductFactory.createProduct(2L));
```

nit : 한번 쓰고 버릴 변수이면서 메소드 호출의 이름만으로도 이미 무엇을 리턴할지 명확하기 떄문에 인라인으로 바꾸는 것이 가독성이 좋아보입니다!